### ISSUE TITLE
Using !Ref in ContentUri of LayerVersion throw error E0001

### ISSUE THREAD

**Issue by kstro21** [aws-cloudformation/cfn-python-lint#1824]
*cfn-lint version: 0.43.0*

*Description of issue.*

The `ContentUri` property of a resource type [AWS::Serverless::LayerVersion][1] can be a string or a `LayerContent` object. But, if we use `!Ref` the template is marked as an error with the message:
>[cfn-lint] E0001: Error transforming template: Resource with id [CommonDependenciesLayer4ffbb56ae8] is invalid. 'ContentUri' requires Bucket and Key properties to be specified.

Here is an example:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Parameters:
  EnvironmentName:
    Description: An environment name
    Type: String
    Default: staging
    AllowedValues:
      - sandbox
      - prod
      - staging

  LayerContentUri:
    Type: String
    Description: Layer content uri
    Default: '../dependencies-layer/nodejs'


Resources:
  CommonDependenciesLayer:
    Type: AWS::Serverless::LayerVersion
    Properties:
      LayerName: !Sub '${EnvironmentName}-common-dependencies'
      Description: 'Common dependencies'
      ContentUri: !Ref LayerContentUri
      CompatibleRuntimes:
        - nodejs12.x
      RetentionPolicy: Retain
    Metadata:
      BuildMethod: nodejs12.x
```
The template deploys ok using `sam deploy` so it is not an error. I'm already using the latest version of `cfn-lint` and already did `cfn-lint -u`

Hope this gets fixed soon.

PS: nice job!

[1]: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-layerversion.html

**Comment by kddejong** [comment]
@kstro21 thanks for reporting this.  I got a fix in that will be released in the next version.

**Comment by kddejong** [comment]
Fixed and released in v0.44.3
